.github/files/gh-autorelease/files/autorelease.sh

```diff
@@ -48,76 +55,78 @@ echo "::group::Creating $SLUG.zip"
 git archive -v --output="$SLUG.zip" --prefix="$SLUG/" HEAD 2>&1
 echo "::endgroup::"
 
-## Create the release note.
-# Extract the changelog section.
-echo "::group::Extracting release notes"
-if [[ ! -f CHANGELOG.md ]]; then
-	echo '::endgroup::'
-	echo '::error::No CHANGELOG.md for release notes.'
-	exit 1
-fi
-SCRIPT="
-	/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
-		bc
+if [[ -z "$ROLLING_MODE" ]]; then
+	## Create the release note.
+	# Extract the changelog section.
+	echo "::group::Extracting release notes"
+	if [[ ! -f CHANGELOG.md ]]; then
+		echo '::endgroup::'
+		echo '::error::No CHANGELOG.md for release notes.'
+		exit 1
+	fi
+	SCRIPT="
+		/^## \\[?$(sed 's/[.\[\]\\*^$\/()+?{}|]/\\&/g' <<<"${TAG#v}")\\]? - / {
+			bc
+			:a
+			n
+			/^## / {
+				q
+			}
+			:c
+			s/^## \[([^]]+)\]/## \1/
+			p
+			ba
+		}
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
+	if [[ -z "$ENTRY" ]]; then
+		echo '::endgroup::'
+		echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
+		exit 1
+	fi
+
+	# Strip unwanted sections.
+	SCRIPT="
 		:a
-		n
-		/^## / {
-			q
+		/^### .* This section will not be copied to readme\.txt/ {
+			:b
+			n
+			/^#/ ba
+			bb
 		}
-		:c
-		s/^## \[([^]]+)\]/## \1/
 		p
-		ba
-	}
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" CHANGELOG.md)
-if [[ -z "$ENTRY" ]]; then
-	echo '::endgroup::'
-	echo "::error::Failed to find section for ${TAG#v} in CHANGELOG.md"
-	exit 1
+	"
+	ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
+
+	echo "Release notes:"
+	echo "-----"
+	echo "$ENTRY"
+	echo "-----"
+	echo "::endgroup::"
+else
+	## Using a brief explanation for the rolling release note.
+	ENTRY="### Rolling release based on the trunk branch."
 fi
 
-# Strip unwanted sections.
-SCRIPT="
-	:a
-	/^### .* This section will not be copied to readme\.txt/ {
-		:b
-		n
-		/^#/ ba
-		bb
-	}
-	p
-"
-ENTRY=$(sed -n -E -e "$SCRIPT" <<<"$ENTRY")
-
-echo "Release notes:"
-echo "-----"
-echo "$ENTRY"
-echo "-----"
-echo "::endgroup::"
+if [[ "true" == "$ROLLING_MODE" ]]; then
+	echo "::group::Deleting stale rolling release"
+	gh release delete "$TAG" --cleanup-tag -y
```

Comment:
You'll need to account for the initial run where no rolling-release release exists yet. With the `-eo pipefail` at the top of the script, it'll abort if `gh` returns a failure code.

The simplest way would be like this, but that'll also suppress any other failures.
```suggestion
	gh release delete "$TAG" --cleanup-tag -y || true
```
so I might do like this instead
```suggestion
	if gh release view "$TAG" >/dev/null; then
		gh release delete "$TAG" --cleanup-tag -y
	fi
```